### PR TITLE
Fixes #35

### LINKED ISSUE #35: Make it simpler to override a template image
The [current method](https://github.com/cisco/elsy/blob/master/docs/templates.md) for overriding templates is a bit cumbersome. It would be nice to add a field to `lc.yml` that allows a user to specify the template image they want to use for that repo.

### PR DESCRIPTION
**Pull request by joeygibson**
This fixes #35.

### CODE REVIEW

File: template/library_test.go [hunk 1]
```diff
@@ -68,16 +69,59 @@ func TestTemplateV1Registration(t *testing.T) {
...
+  } else {
+    if (strings.Contains(tmplyaml, "image: bazimage")) {
+      t.Error("template should not have had image in it")
+    }
+  }
+
+  if tmplyaml, err := GetV1("baz", false, "bazimage"); err != nil {
+    t.Error("expected Get to return yaml after registering a template")
+  } else {
+    if (strings.Contains(tmplyaml, "image: bazimage")) {
```
Comment (paulcichonski) [inline template/library_test.go]: How does this test pass, doesn't the template_image interpolation logic rely on the `template.composeYmlTmpl` having `{{.TemplateImage}}` inside it?

Comment (joeygibson) [inline template/library_test.go]: That field it still there, it's just an empty string. And in the template logic, an empty string evaluates to false.

**Comment by paulcichonski** [comment]
Do you mind updating these examples to use the new and improved image override logic:

- https://github.com/cisco/elsy/blob/master/examples/java-library/docker-compose.yml
- https://github.com/cisco/elsy/blob/master/examples/java-note-service/docker-compose.yml#L16

**Comment by paulcichonski** [comment]
other than the above comments this LGTM, much improved change!

Also, probably need a blurb in the `CHANGELOG.md` documenting this change.

**Pull request merged**